src/copy.rs

```diff
@@ -598,8 +627,12 @@ pub(crate) fn copy_internal(options: Options<'_>,
     let (queue, clipboard_manager, devices) = get_devices(primary, seat, socket_name)?;
 
     // Collect the source data to copy.
-    let (mime_type, data_path) =
-        make_source(source, mime_type, trim_newline).map_err(Error::TempCopy)?;
+    let mut data_paths = HashMap::new();
+    for mimesource in sources {
+        let (mime_type, data_path) =
+            make_source(mimesource.source, mimesource.mime, trim_newline).map_err(Error::TempCopy)?;
```

Comment:
If any of these operations fails, it should clean up files the previous ones. The nicest way to go about that is to make it return a RAII guard for the temp file instead of just a path.

Reply:
Thank'you, I'll need some time to look how to do it, as I'm very new to rust.